src/libraries/Microsoft.PowerFx.Json/Functions/JsonFunctionImpl.cs

```diff
@@ -269,6 +269,21 @@ public void Visit(UntypedObjectValue untypedObjectValue)
                 {
                     throw new ArgumentException($"Unable to serialize type {untypedObjectValue.GetType().FullName} to Json format.");
                 }
+
+                public void Visit(BlobValue value)
+                {
+                    throw new NotImplementedException();
+                }
+
+                public void Visit(MediaValue value)
+                {
+                    throw new NotImplementedException();
```

Comment:
>throw new NotImplementedException(); [](http://example.com/codeflow?start=20&length=36)

Not initially, but we should consider adding support for serializing media/images to JSON: this is available in Power Apps, where it creates a datauri,base64-encoded version of it. We should think about guards for that (in Power Apps this is only available in behavior properties) to prevent makers from shooting themselves in the foot by running such expensive operations.

Reply:
Also required for connectors, I'll add that tomorrow or next week